WordPress/gutenberg
# Font Library: add a setting to customize the conditional download of font assets from a font collection

## What?
Adds a setting to customize the conditional download of font assets from a font collection.

This is a simpler approach of https://github.com/WordPress/gutenberg/pull/57989

## Why?
To allow extenders to customize if WordPress Font Library will download the assets when installing font faces from a font collection.

## How?
Adds the `fontLibraryDownloadAssets` editor setting that can be customized with PHP using a filter (See testing instructions).
It adds `true` as the default value for this setting.

## Testing Instructions
1. Check that the font library continues to work as expected. By default, the assets from the font families/faces installed from a font collection featuring external links for the font face assets should be downloaded to the fonts folder (`wp-content/fonts` by default). So if you install font families from the Google Fonts collection, you should have the corresponding font assets downloaded in your fonts folder (`wp-content/fonts`).

2. Use this PHP filter to disable the download of the font assets of the font families installed. With this filter, when installing font faces from a font collection (like the default Google Fonts collection) the font asset won't be downloaded to the fonts folder (`wp-content/fonts`), instead of that, the fonts will continue being served from the link provided in the font face definition.

```php
function disable_font_library_downloads( $editor_settings ) { 
	$editor_settings['fontLibraryDownloadAssets'] = false;
    return $editor_settings; 
}

add_filter( "block_editor_settings_all", "disable_font_library_downloads" );
```
